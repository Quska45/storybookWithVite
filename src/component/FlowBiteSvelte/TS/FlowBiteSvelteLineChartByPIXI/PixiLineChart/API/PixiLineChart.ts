import type { Application } from "pixi.js";
import type { ContainerManager } from "../Container/ContainerManager";

export class PixiLineChart {
    app: Application;
    labels;
    datasets;
    containerManager: ContainerManager;
    width;
    height;
    verticalMargin;
    horizontalMargin;
    

    constructor( app: Application, containerManager: ContainerManager ){
        this.app = app;
        this.labels = [];
        this.datasets = [];
        this.containerManager = containerManager;
    };

    attachBySelector( selector: string ){
        let dom = (document.querySelector( selector ) as HTMLElement);
        let view: HTMLCanvasElement = ( this.app.view as HTMLCanvasElement );

        dom.appendChild( view );
    };

    
}