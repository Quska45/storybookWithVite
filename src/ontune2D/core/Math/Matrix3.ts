export class Matrix3 {
    elements: number[];

    constructor(){
        this.elements = [
            1, 0, 0,
            0, 1, 0,
            0, 0, 1
        ];
    };

    setAll( m11, m12, m13, m21, m22, m23, m31, m32, m33 ){
        this.elements = [
            
            m11, m12, m13,
            m21, m22, m23,
            m31, m32, m33
            
        ];
        return this;
    };
}