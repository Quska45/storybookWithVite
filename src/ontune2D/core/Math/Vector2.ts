export class Vector2{
    x: number;
    y: number;

    constructor( x: number, y: number ){
        this.x = x;
        this.y = y;
    };

    set(  x: number, y: number  ){
        const vector2 = this;

        vector2.x = x;
        vector2.y = y;
    };
};