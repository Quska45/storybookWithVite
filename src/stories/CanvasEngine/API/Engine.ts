import { Renderer } from "../Core/Basic/Renderer";
import type { Vector } from "../Core/Math/Vector";
import type { Object2D } from "../Core/Object2D/Object2D";
import { Scene } from "../Core/Object2D/Scene";
import { Line } from "../Core/Object2D/Shape/Line";
import { Text } from "../Core/Object2D/Shape/Text";

export class Engine {
    renderer: Renderer;
    scene: Scene;
    RAF = {
        rendering: null,
        fpms: -1
    };
    
    constructor( selector: string ){
        this.renderer = new Renderer( selector );
        this.scene = new Scene( 'Scene' + selector );
    };

    run(){
        let _this = this;
        (function run( time ){
            _this.RAF.rendering = requestAnimationFrame( run );
            _this.RAF.fpms = time;
            _this.renderer.render( _this.scene );
        })()
    };

    stop(){
        cancelAnimationFrame( this.RAF.rendering );
        this.RAF.rendering = null;
    };

    addLine( id: string, firstCoordinate: Vector, secondCoordinate: Vector ): Object2D | null{
        let object2d = this.getObjectById( id );
        if( object2d ){
            console.log( `'${ id }'는 중복된 id 값 입니다.` );
            return null;
        };

        let line = new Line( id );
        this.scene.add( line );

        return line;
    };

    addText( id: string, textStr: string, position: Vector ): Object2D | null{
        let object2d = this.getObjectById( id );
        if( object2d ){
            console.log( `'${ id }'는 중복된 id 값 입니다.` );
            return null;
        };

        let text = new Text( id, textStr );
        this.scene.add( text );

        return text;
    };

    getObjectById( id: string ): Object2D{
        let targetObject2d = this.scene.getObjectById( id );
        return targetObject2d ? targetObject2d : null;
    };
};