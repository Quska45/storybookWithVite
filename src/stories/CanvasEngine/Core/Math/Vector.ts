export class Vector {
    x: number | undefined = 0;
    y: number | undefined = 0;

    constructor( x?:number, y?:number ){
        this.x = x;
        this.y = y;
    };
}