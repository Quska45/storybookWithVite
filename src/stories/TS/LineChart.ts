import type { Engine } from "../CanvasEngine/API/Engine"

export class LineChart {
    engine: Engine;

    constructor( engine: Engine ){
        this.engine = engine;
    };

    
};