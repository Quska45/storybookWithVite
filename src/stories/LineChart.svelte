<script lang="ts">
    import { Engine } from './CanvasEngine/API/Engine';
    import { onMount } from 'svelte';

    onMount(async () => {
        let engine = new Engine( '#canvasTarget' );
    });  
</script>
<div id='canvasTarget' style="width: 300px; height: 300px; border: 1px solid red;">
</div>
