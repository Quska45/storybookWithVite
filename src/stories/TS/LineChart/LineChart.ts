import { Engine } from "../../CanvasEngine/API/Engine"
import { Vector } from "../../CanvasEngine/Core/Math/Vector";
import { Quadrant } from "../../CanvasEngine/Core/Object2D/Scene";
import type { TConfig } from "../Common/Config";

export class LineChart {
    engine: Engine;
    config: TConfig;

    constructor( selector, config: TConfig ){
        this.engine = new Engine( selector );
        this.config = config;
    };

    init(){
        this.engine.setQuadrant( Quadrant.quadrant1 );
        this.engine.addLine( 'testLine', new Vector(0,0), new Vector(50,50) );
        this.engine.addText('1', 'abcdef', new Vector(50,50));
        // engine.renderer.ctx.restore()
    };

    run( callback? ){
        this.engine.run();
    }
};