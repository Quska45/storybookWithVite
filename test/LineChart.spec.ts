import { render, screen, fireEvent } from '@testing-library/svelte';
// import App from '../src/App.svelte';
import LineChart from '../src/stories/LineChart.svelte'
import { Engine } from '../src/stories/CanvasEngine/API/Engine'
import { Vector } from '../src/stories/CanvasEngine/Core/Math/Vector';

describe('LineChart Container load test', ()=>{
    // test("LineChart load test by pure screen object", () => {
    //     render(LineChart);
    //     // const node = screen.queryByText("Hello world!");
    //     // expect(node).not.toBeNull();
    //     // testing library를 사용하기 위해 실제 코드에 role을 추가해야 하는 것은 이상한 것 같다.
    //     let node = screen.getByRole( 'canvasTarget' );
    //     let dom = document.getElementById('canvasTarget');
    //     expect(node).toBeInTheDocument();
    // });
    
    test("LineChart load test by dom", () => {
        let lineChart = render(LineChart);
        let canvasTarget = lineChart.container.querySelector( '#canvasTarget' );
    
        expect(canvasTarget).toBeInTheDocument();
    });
});

describe('canvas load test', () => {
    test("canvas load test by dom", () => {
        let lineChart = render(LineChart);
        let canvas = lineChart.container.querySelector( 'canvas' );
    
        expect(canvas).toBeInTheDocument();
    });

    test('canvas draw test', async () => {
        let lineChart = render(LineChart);
        let engine = new Engine( '#canvasTarget' );

        engine.addLine( 'testLine', new Vector(1,1), new Vector(5,5) );
        engine.run();
    
        // expect(canvas).toBeInTheDocument();
    });
});

