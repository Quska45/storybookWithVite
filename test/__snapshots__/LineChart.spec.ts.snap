// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`canvas load test canvas draw test 1`] = `
[
  {
    "props": {
      "height": 150,
      "width": 300,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "value": "#000000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  {
    "props": {
      "value": "#000000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  {
    "props": {
      "value": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  {
    "props": {
      "x": 2,
      "y": 2,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "moveTo",
  },
  {
    "props": {
      "x": 5,
      "y": 5,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineTo",
  },
  {
    "props": {
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "x": 2,
            "y": 2,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        {
          "props": {
            "x": 5,
            "y": 5,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "closePath",
  },
]
`;
